Code/Framework/AzFramework/Platform/Linux/AzFramework/Windowing/NativeWindow_Linux_xcb.cpp

```diff
@@ -0,0 +1,244 @@
+
+        xcb_check_result = xcb_create_window_checked(m_xcbConnection,
+                                                     XCB_COPY_FROM_PARENT,
+                                                     m_xcbWindow,
+                                                     xcbParentWindow,
+                                                     aznumeric_cast<int16_t>(geometry.m_posX),
+                                                     aznumeric_cast<int16_t>(geometry.m_posY),
+                                                     aznumeric_cast<int16_t>(geometry.m_width),
+                                                     aznumeric_cast<int16_t>(geometry.m_height),
+                                                     borderWidth,
+                                                     XCB_WINDOW_CLASS_INPUT_OUTPUT,
+                                                     xcbRootScreen->root_visual,
+                                                     eventMask,
+                                                     valueList);
+
+        AZ_Assert(ValidateXcbResult(xcb_check_result), "Failed to create xcb window.");
+
+        SetWindowTitle(title);
+
+        // Setup the window close event 
+        const static char* wmProtocolString = "WM_PROTOCOLS";
+        
+        xcb_intern_atom_cookie_t cookie_protocol = xcb_intern_atom(m_xcbConnection, 1, strlen(wmProtocolString), wmProtocolString);
+        xcb_intern_atom_reply_t* reply_protocol = xcb_intern_atom_reply(m_xcbConnection, cookie_protocol, nullptr);
+        AZ_Error("AtomVulkan_RHI", reply_protocol!=NULL, "Unable to query xcb '%s' atom", wmProtocolString);
+        m_xcbAtomProtocols = reply_protocol->atom;
+
+        const static char* wmDeleteWindowString = "WM_DELETE_WINDOW";
+        xcb_intern_atom_cookie_t cookie_delete_window = xcb_intern_atom(m_xcbConnection, 0, strlen(wmDeleteWindowString), wmDeleteWindowString);
+        xcb_intern_atom_reply_t* reply_delete_window = xcb_intern_atom_reply(m_xcbConnection, cookie_delete_window, nullptr);
+        AZ_Error("AtomVulkan_RHI", reply_delete_window!=NULL, "Unable to query xcb '%s' atom", wmDeleteWindowString);
+        m_xcbAtomDeleteWindow = reply_delete_window->atom;
+
+        xcb_check_result = xcb_change_property_checked(m_xcbConnection, 
+                                                       XCB_PROP_MODE_REPLACE, 
+                                                       m_xcbWindow, 
+                                                       m_xcbAtomProtocols, 
+                                                       XCB_ATOM_ATOM, 
+                                                       s_XcbFormatDataSize, 
+                                                       1,
+                                                       &m_xcbAtomDeleteWindow);
+                                                    
+        AZ_Assert(ValidateXcbResult(xcb_check_result), "Failed to change atom property for WM_CLOSE event");
```

Comment:
atom property?

Reply:
Yeah, and xcb construct. Not to be confused with Atom

Reply:
Ah I see, missed that it was an xcb thing!